src/osd/ECUtil.cc

```diff
@@ -12,11 +12,11 @@ using ceph::ErasureCodeInterfaceRef;
 using ceph::Formatter;
 
 std::pair<uint64_t, uint64_t> ECUtil::stripe_info_t::chunk_aligned_offset_len_to_chunk(
-  std::pair<uint64_t, uint64_t> in) const {
+  uint64_t off, uint64_t len) const {
   pair<uint64_t, uint64_t> tmp = offset_len_to_stripe_bounds(in);
   return std::make_pair(
-    chunk_aligned_logical_offset_to_chunk_offset(tmp.first),
-    chunk_aligned_logical_size_to_chunk_size(tmp.second));
+    chunk_aligned_logical_offset_to_chunk_offset(off),
```

Comment:
You've change the function and regressed one of the partial read fixes (although presumably later commits address this) - old code took "logical" offset/length parameters and realigned these to a stripe boundary before converting these to "chunk_offsets' in the shard. The change removes the realignment to a stripe boundary which means that this function will not always return a sufficient range of the shard, it also makes tmp an unused variable

For example stripe size = 8K, offset = 4K, length = 8K will return shard offset 0, length 4K which is not enough.

Reply:
The fixes put in before to this function fixed up the old code-path, but broke the new one, which relied on the new behaviour.  I will have a think about how to address your concerns... Maybe I end up refactoring this out completely.

Reply:
This is actually fixed in a later commit (abae77baad008f211803722cef6122422796912b). It is refactored to not use pairs, but functionally the same as the code you wrote.